Add options to confirm reblog/unreblog actions, close #460

charlag opened the pull request:
Add an option to confirm reblogs.
Initially I wanted to put it much further the pipeline but to get animations work right I had to work around `SparkButton` a little bit. The advantage is that it's implemented in just one place now.

It tries to mimick Masto UI but not very closely.

![image](https://user-images.githubusercontent.com/3099142/72672087-66d3a900-3a55-11ea-9fe1-e4db6a7901da.png)

Tak commented on app/src/main/java/com/keylesspalace/tusky/adapter/StatusBaseViewHolder.java:
```diff
@@ -601,12 +605,25 @@ protected void setupButtons(final StatusActionListener listener, final String ac
+                    int okButtonTextId =
+                            buttonStatus ? R.string.action_reblog : R.string.action_unreblog;
+                    new AlertDialog.Builder(reblogButton.getContext())
+                            .setMessage(statusContent)
```
Shouldn't this be clamped to some maximum length? Or does the dialog do this automatically?

charlag replied:
I checked and dialog just becomes scrollable but it's a good point

connyduck commented on app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.java:
```diff
@@ -576,6 +577,16 @@ public void onReply(int position) {
+        boolean confirmReblogs = PreferenceManager.getDefaultSharedPreferences(getContext())
+                .getBoolean("confirmReblogs", false);
+        if (confirmReblogs) {
+
```
looks like something is mising here

connyduck commented on app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.java:
```diff
@@ -576,6 +577,16 @@ public void onReply(int position) {
+                .getBoolean("confirmReblogs", false);
+        if (confirmReblogs) {
+
+        } else {
```
looks like something is mising here

charlag replied:
I find it funny that you missed that 's' first :laughing:

connyduck replied:
:D

connyduck commented:
Probably best to add a return value to the `onEvent` of the `SparkEventListener` so you can return false when you don't want the animation to play

connyduck commented:
Here, I made the change to `SparkButton`: https://github.com/connyduck/SparkButton/pull/2

You can try it out with ` implementation "com.github.connyduck:sparkbutton:1e0d112310"`

charlag commented:
Thanks, it works now (even though it's a breaking change to the library).
I didn't find a way to play animation at first but then I added it. Works fine but you can still break it but triggering it back and forth. Go figure.

I would squash it all into one commit before merging, I just split it so that you see the diff

connyduck commented on app/src/main/java/com/keylesspalace/tusky/view/ConditionalSparkButton.java:
```diff
@@ -0,0 +1,41 @@
+
+import at.connyduck.sparkbutton.SparkButton;
+
+public class ConditionalSparkButton extends SparkButton {
```
this class is no longer required

charlag commented:
>ERROR: Failed to resolve: com.github.connyduck:sparkbutton:4.0.0

connyduck commented:
try again, build wasn't finished

charlag commented:
Updated, squashed, rebased (that card thing lead to some headache with commas)
